### PR TITLE
Map cycling button widget and cycling options

### PR DESCRIPTION
**Pull request by liach**
Turns out there are still like 2 other inner classes, but I will get em

Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

**Review by YanisBft — APPROVED** [review]

**Comment by liach** [comment]
Requesting reviews. Imo this introduces some javadoc and makes cycling options much easier for mods to use, and only renames `getGenericTextForValue`, which should have a small impact.

File: mappings/net/minecraft/client/gui/widget/CyclingButtonWidget.mapping [hunk 1]
```diff
@@ -1,32 +1,164 @@
...
+		ARG 13 tooltipFactory
+		ARG 14 optionTextOmitted
 	METHOD method_32603 getValue ()Ljava/lang/Object;
 	METHOD method_32604 (Ljava/lang/Boolean;)Lnet/minecraft/class_2561;
 		ARG 0 value
 	METHOD method_32605 setValue (Ljava/lang/Object;)V
 		ARG 1 value
+	METHOD method_32606 builder (Ljava/util/function/Function;)Lnet/minecraft/class_5676$class_5677;
+		COMMENT Creates a new builder for a cycling button widget.
+		ARG 0 messageFactory
```
Comment (enbrain) [inline mappings/net/minecraft/client/gui/widget/CyclingButtonWidget.mapping]:
```suggestion
		ARG 0 valueToText
```
I don't know if you intentionally avoided `valueToText` here.

File: mappings/net/minecraft/client/gui/widget/CyclingButtonWidget.mapping [hunk 2]
```diff
@@ -1,32 +1,164 @@
...
+			COMMENT {@link CyclingButtonWidget#getGenericNarrationMessage()} for narration
+			COMMENT messages.
+			ARG 1 narrationMessageFactory
+		METHOD method_32624 values ([Ljava/lang/Object;)Lnet/minecraft/class_5676$class_5677;
+			COMMENT Sets the option values for this button.
+			ARG 1 values
 		METHOD method_32625 (Ljava/lang/Object;)Ljava/util/List;
 			ARG 0 value
-	CLASS class_5678
+	CLASS class_5678 Callback
```
Comment (enbrain) [inline mappings/net/minecraft/client/gui/widget/CyclingButtonWidget.mapping]:
```suggestion
	CLASS class_5678 ValueChangeListener
```
Calling it `Callback` is too generic IMO.

File: mappings/net/minecraft/client/option/CyclingOption.mapping [hunk 3]
```diff
@@ -22,25 +29,34 @@ CLASS net/minecraft/class_4064 net/minecraft/client/option/CyclingOption
...
 		ARG 3 getter
 		ARG 4 setter
 	METHOD method_32527 (Ljava/util/List;Ljava/lang/Boolean;)Ljava/util/List;
 		ARG 1 value
+	METHOD method_32528 tooltip (Ljava/util/function/Function;)Lnet/minecraft/class_4064;
+		ARG 1 tooltips
 	METHOD method_32532 (Lnet/minecraft/class_2561;Lnet/minecraft/class_310;)Lnet/minecraft/class_5676$class_5679;
 		ARG 1 client
-	CLASS class_5675
+	CLASS class_5675 Consumer
```
Comment (enbrain) [inline mappings/net/minecraft/client/option/CyclingOption.mapping]:
```suggestion
	CLASS class_5675 Setter
```
This interface is used exclusively for setters.

File: mappings/net/minecraft/client/gui/widget/CyclingButtonWidget.mapping (hunk 2 above)
Comment (liach) [reply mappings/net/minecraft/client/gui/widget/CyclingButtonWidget.mapping]: imo callback suffices as callbacks are for updates, which is what this thing does

File: mappings/net/minecraft/client/option/CyclingOption.mapping (hunk 3 above)
Comment (liach) [reply mappings/net/minecraft/client/option/CyclingOption.mapping]: actually you are right, this name is more concise as well!

File: mappings/net/minecraft/client/gui/widget/CyclingButtonWidget.mapping (hunk 2 above)
Comment (liach) [reply mappings/net/minecraft/client/gui/widget/CyclingButtonWidget.mapping]:
```suggestion
	CLASS class_5678 UpdateCallback
```

**Review by enbrain — APPROVED** [review]

**Pull request merged**